auth/iauthenticator.go

```diff
@@ -0,0 +1,8 @@
+// Copyright 2023 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package auth
+
+type IAuthenticator interface {
+	Configure(cfg map[string]interface{}) error
```

Comment:
I think, we should have a `Validate` Method on the IAuthenticator Interface

Reply:
To what end? Validation is performed as part of Configure() which is the only way for the user to alter the state of an `IAuthenticator`

Reply:
True that this is part of Configure that you call Validator.
I am ok with this!